src/abstractdataframe/join.jl

```diff
@@ -430,23 +345,622 @@ function Base.join(df1::AbstractDataFrame, df2::AbstractDataFrame;
+
+# Keyword Arguments
+- `on` : A column name to join `df1` and `df2` on. If the columns on which
+  `df1` and `df2` will be joined have different names, then a `left=>right`
+  pair can be passed. It is also allowed to perform a join on multiple columns,
+  in which case a vector of column names or column name pairs can be passed
+  (mixing names and pairs is allowed). If more than two data frames are joined
+  then only a column name or a vector of column names are allowed.
+  `on` is a required argument.
+- `makeunique` : if `false` (the default), an error will be raised
+  if duplicate names are found in columns not joined on;
+  if `true`, duplicate names will be suffixed with `_i`
+  (`i` starting at 1 for the first duplicate).
+- `indicator` : Default: `nothing`. If a `Symbol`, adds categorical indicator
+   column named `Symbol` for whether a row appeared in only `df1` (`"left_only"`),
+   only `df2` (`"right_only"`) or in both (`"both"`). If `Symbol` is already in use,
+   the column name will be modified if `makeunique=true`.
+   This argument is only supported when joining exactly two data frames.
+- `validate` : whether to check that columns passed as the `on` argument
+   define unique keys in each input data frame (according to [`isequal`](@ref)).
+   Can be a tuple or a pair, with the first element indicating whether to
+   run check for `df1` and the second element for `df2`.
+   By default no check is performed.
+
+When merging `on` categorical columns that differ in the ordering of their levels, the
+ordering of the left data frame takes precedence over the ordering of the right data frame.
+
+If more than two data frames are passed, the join is performed
+recursively with left associativity.
+In this case the `indicator` keyword argument is not supported.
+
+# Examples
+```julia
+julia> name = DataFrame(ID = [1, 2, 3], Name = ["John Doe", "Jane Doe", "Joe Blogs"])
+3×2 DataFrame
+│ Row │ ID    │ Name      │
+│     │ Int64 │ String    │
+├─────┼───────┼───────────┤
+│ 1   │ 1     │ John Doe  │
+│ 2   │ 2     │ Jane Doe  │
+│ 3   │ 3     │ Joe Blogs │
+
+julia> job = DataFrame(ID = [1, 2, 4], Job = ["Lawyer", "Doctor", "Farmer"])
+3×2 DataFrame
+│ Row │ ID    │ Job    │
+│     │ Int64 │ String │
+├─────┼───────┼────────┤
+│ 1   │ 1     │ Lawyer │
+│ 2   │ 2     │ Doctor │
+│ 3   │ 4     │ Farmer │
+
+julia> innerjoin(name, job, on = :ID)
+2×3 DataFrame
+│ Row │ ID    │ Name     │ Job    │
+│     │ Int64 │ String   │ String │
+├─────┼───────┼──────────┼────────┤
+│ 1   │ 1     │ John Doe │ Lawyer │
+│ 2   │ 2     │ Jane Doe │ Doctor │
+
+julia> job2 = DataFrame(identifier = [1, 2, 4], Job = ["Lawyer", "Doctor", "Farmer"])
+3×2 DataFrame
+│ Row │ identifier │ Job    │
+│     │ Int64      │ String │
+├─────┼────────────┼────────┤
+│ 1   │ 1          │ Lawyer │
+│ 2   │ 2          │ Doctor │
+│ 3   │ 4          │ Farmer │
+
+julia> innerjoin(name, job2, on = [:ID => :identifier])
```

Comment:
Using a vector with a single value could mislead users into thinking that a pair can only be used with a vector. Same below.
```suggestion
julia> innerjoin(name, job2, on = :ID => :identifier)
```